plexapi/library.py

```diff
@@ -436,21 +445,83 @@ def reload(self, key=None):
                 kwargs (dict): Dict of settings to edit.
         """
         if not agent:
             agent = self.agent
-        part = '/library/sections/%s?agent=%s&%s' % (self.key, agent, urlencode(kwargs))
+
+        locations = []
+        if kwargs['location']:
+            for path in kwargs['location']:
+                if not self._server.isBrowsable(path):
+                    raise BadRequest('Path: %s does not exist.' % path)
+            locations = [('location', path) for path in kwargs.pop('location')]
+
+        params = list(kwargs.items()) + locations
+
+        part = '/library/sections/%s?agent=%s&%s' % (self.key, agent, urlencode(params, doseq=True))
         self._server.query(part, method=self._server._session.put)
 
         # Reload this way since the self.key dont have a full path, but is simply a id.
         for s in self._server.library.sections():
             if s.key == self.key:
                 return s
 
+    def addLocations(self, location):
+        """ Add a location to a library.
+
+            Parameters:
+                location (str, list, or :class:~plexapi.library.Path)): A single folder path, list of paths,
+                    or :class:`~plexapi.library.Path`.
+
+            Example:
+
+                .. code-block:: python
+                LibrarySection.addLocations('/path/1')
+                LibrarySection.addLocations(['/path/1', 'path/2', '/path/3'])
+                LibrarySection.addLocations(PlexServer.browse()[0])
+        """
+        locations = self.locations
+        if isinstance(location, Path):
+            location = location.path
+        if isinstance(location, list):
+            locations.extend(location)
+        else:
+            locations.append(location)
+        for path in locations:
+            if not self._server.isBrowsable(path):
+                raise BadRequest('Path: %s does not exist.' % path)
+        self.edit(location=locations)
+
+    def removeLocations(self, location):
+        """ Remove a location from a library.
+
+            Parameters:
+                location (str, list, or :class:~plexapi.library.Path): A single folder path, list of paths,
+                     or :class:`~plexapi.library.Path`.
+            Example:
+
+                .. code-block:: python
+                LibrarySection.removeLocations('/path/1')
+                LibrarySection.removeLocations(['/path/1', 'path/2', '/path/3'])
+                LibrarySection.removeLocations(PlexServer.browse()[0])
+        """
+        locations = self.locations
+        if isinstance(location, Path):
+            location = location.path
+        if isinstance(location, list):
+            locations = [path for path in locations if path not in location]
+        else:
+            locations.remove(location)
+        for path in locations:
+            if not self._server.isBrowsable(path):
+                raise BadRequest('Path: %s does not exist.' % path)
```

Comment:
The exception should be something like "The path you are trying to remove does not exist in the library". Presumably the current paths in the library are already browsable and don't need to be checked when removing.

Reply:
d1b1903 and 5cf95e5